maturity-level-two/src/Services/CoditoRepository.cs

```diff
@@ -76,30 +78,62 @@ public async Task CreateCustomizationAsync(Customization customization)
 
             foreach (var property in properties)
             {
-                _coditoContext.Entry(entityToUpdate).Property(property).IsModified = true;
+                if(!_coditoContext.Entry(entityToUpdate).Property(property).Metadata.IsKey())
+                {
+                    _coditoContext.Entry(entityToUpdate).Property(property).IsModified = true;
+                }               
             }
 
             await _coditoContext.SaveChangesAsync();
         }
 
-        public async Task DeleteCustomizationAsync(int id)
+        public async Task<int> DeleteCustomizationAsync(int id)
         {
-            var entityToDelete = _coditoContext.Customizations.Find(id);
-            _coditoContext.Customizations.Remove(entityToDelete);
-            await _coditoContext.SaveChangesAsync();
-        }
 
-        public async Task ApplyCustomizationSaleAsync(Customization customization)
-        {
-            //var entity = await _coditoContext.Customizations.Where(cust => cust.Id == id).FirstOrDefaultAsync();
-            customization.InventoryLevel --;
-            customization.NumberSold++;
+            bool exists = await CustomizationExistsAsync(id);
 
-            _coditoContext.Customizations.Update(customization);
-
-            await _coditoContext.SaveChangesAsync();
+            if (exists)
+            {
+                var entityToDelete = new Customization();
+                Type type = entityToDelete.GetType();
+                PropertyInfo prop = type.GetProperty("Id");
+                prop.SetValue(entityToDelete, id, null);
```

Comment:
Does this work without having to specify `BindingFlags.Instance `and `BindingFlags.NonPublic` , since the setter is private ?

Reply:
This reflection method is new to me, but yes, it appears the work, apparently because the property itself is public